File: apps/mobile/src/types/analyze.ts
export interface PlatformPrice {
  name: string;
  price: number;
  url: string;
}

export interface AnalyzeResponseDto {
  name: string;
  category: string;
  condition: string;
  estimatedPrice: number;
  priceRange: {
    min: number;
    max: number;
  };
  confidence: number;
  platforms: PlatformPrice[];
  tips: string[];
}

File: apps/api/app/schemas/analyze.py
from pydantic import BaseModel
from typing import Any


class PlatformPrice(BaseModel):
    name: str
    price: float
    url: str


class AnalyzeResponse(BaseModel):
    name: str
    category: str
    condition: str
    estimatedPrice: float
    priceRange: dict[str, float]
    confidence: float
    platforms: list[PlatformPrice]
    tips: list[str]
